### ISSUE TITLE
Softmax scores of Cross-encoder tending towards 0 or 1

### ISSUE THREAD

**Issue by horiacristescu** [UKPLab/sentence-transformers#1273]
I am training a Cross Encoder with 30K examples for 20 epochs and I find the softmax prob scores going too much towards 0 or 1. Basically 99% of the values are >0.9 or <0.1 and few in between. 

Is this expected behaviour? If you compute a triplet margin with this model the margin is going to be -1, 0 or 1 most of the time.

**Comment by nreimers** [comment]
Yes, this is normal for any classification model that is trained with Softmax and CrossEntropyLoss. Labels are usually 0 or 1, so models have a strong tendency to either predict scores close to 1 or scores close to 0.

If you need labels in between, you must either have training data with labels in between or use a bi-encoder. There, the distance in the vector space is more meaningful
